### ISSUE TITLE
Multi-currency support for promotion conditions & actions

### ISSUE THREAD

**Issue by StampixSMO** [vendurehq/vendure#2520]
**Describe the bug**
When using channels with multi-currency, creating promotions is not very straightforward because only the default currency is supported.
This results in potential unexpected discounts, since you can't define a monetary value per currency.

**To Reproduce**
1. Create a new channel with more than 1 currency (and a default one), see my first screenshot
2. Try to create a promotion with a condition or action that requires a monetary value, see my second screenshot.
3. It's not possible to define the monetary value in another currency

![image](https://github.com/vendure-ecommerce/vendure/assets/79153163/55e8e845-bc7a-4bcd-a996-2f253a067d2e)
![image](https://github.com/vendure-ecommerce/vendure/assets/79153163/de3eb4b9-4158-408b-b770-5daabec57b66)

**Expected behavior**
The promotion condition & action to be multi-currency supportive, i.e. you can enter the monetary value in multiple currencies and the condition/action will act accordingle.

**Environment (please complete the following information):**
 - @vendure/core version: 2.1.2
 - Nodejs version: 18
 - Database (mysql/postgres etc): postgres

**Comment by michaelbromley** [comment]
Hi! Have you tried implementing something like this yourself as a custom PromotionAction? I think it should be possible using existing APIs.

**Comment by StampixSMO** [comment]
I have not tried this yet since we didn't need it just yet, rather I just noticed it wasn't possible.

A custom PromotionAction with a custom input field too then? Which does allow multiple currencies? Since the current one hardcodes the channel's default currency.
Doesn't it make more sense to add this to the existing actions?

**Comment by michaelbromley** [comment]
Yes, it can be added to the default actions, but I suggested it as something you can do if you need support right now and cannot wait for core support. I'm not 100% certain of the best implementation for this, but you could eg define a string list arg which can be used to encode currencyCode/price pairs which are then used by the action to select the correct discount.

**Comment by StampixSMO** [comment]
> Yes, it can be added to the default actions, but I suggested it as something you can do if you need support right now and cannot wait for core support. I'm not 100% certain of the best implementation for this, but you could eg define a string list arg which can be used to encode currencyCode/price pairs which are then used by the action to select the correct discount.

Ah okay, yep that's definitely possible with a custom promotion action as a workaround for the time being. Thanks for the suggestion!

**Comment by dlhck** [comment]
This feature request has been moved to GitHub Discussions for community voting.
Please upvote the discussion if you would like to see this implemented: https://github.com/vendurehq/vendure/discussions/4275
